docs/openapi_generator/pyopenapi/utility.py

```diff
@@ -132,7 +132,18 @@ def _validate_api_method_return_type(method) -> str | None:
 
     return_type = hints['return']
     if is_optional_type(return_type):
-        return "returns Optional type"
+        return "returns Optional type where a return value is mandatory"
+
+
+def _validate_api_method_doesnt_return_list(method) -> str | None:
+    hints = get_type_hints(method)
+
+    if 'return' not in hints:
+        return "has no return type annotation"
+
+    return_type = hints['return']
+    if get_origin(return_type) is list:
+        return "returns a list where a List*Response object is expected"
```

Comment:
Now we expect a PaginatedResponse?

Reply:
@leseb would it be correct only once we enforce paginatedResponse for all relevant GETs? There are still a bunch of APIs that do not have pagination enabled. I'll mention both and we can remove the List*Response when this happens?